### ISSUE TITLE
How to initialize gamma

### ISSUE THREAD

**Issue by peter943** [suhaslohit/TTN#1]
HI
    I have read your paper. I tried to integrate the TTN module into my network to improve the results. But I do not know how to initizalize gamma.  Random initialization is ok? Could you please give me some advices. Thank you very much for your kind help.

**Comment by suhaslohit** [comment]
Hello Peter, thanks for your interest in our work.

My guess is random initialization should be fine. However, in the paper, we
initialized the weights of the TTN such that the initial gamma, at the
beginning of training, was the identity function.

**Comment by peter943** [comment]
> Hello Peter, thanks for your interest in our work.
> 
> My guess is random initialization should be fine. However, in the paper, we
> initialized the weights of the TTN such that the initial gamma, at the
> beginning of training, was the identity function.
> …
Thank you for your help. I study your code. The function warp(pose, gamma) is used to warp the time series. From the figure 1 in your paper, I found that the TTN module contains four steps: Trainable layers, constraint satisfaction, predicted warps and warping.  So, I think the function warp is used to achieve the last two steps. I should achieve the first two steps by myself. And then, put the output into the function warp. I do not know the understanding is right? Could you please give me some help. Thanks again for your kind help!  
                                                                                                                                             Peter

**Comment by suhaslohit** [comment]
Yes, so far, I have only uploaded the warping code. The TTN module should first output a vector, which is then transformed into a warping function. This is shown in Equation 4 of the paper.

I will try and upload a simple training code showing the constraint satisfaction layers in the next few days.

**Comment by peter943** [comment]
> Yes, so far, I have only uploaded the warping code. The TTN module should first output a vector, which is then transformed into a warping function. This is shown in Equation 4 of the paper.
> 
> I will try and upload a simple training code showing the constraint satisfaction layers in the next few days.
Thanks for your help. It is very useful for my present work.

**Comment by suhaslohit** [comment]
I have uploaded the training and testing code for a synthetic dataset. network_ttn_synthetic.py contains the constraint satisfaction layers.

**Comment by peter943** [comment]
> I have uploaded the training and testing code for a synthetic dataset. network_ttn_synthetic.py contains the constraint satisfaction layers.

Thanks a lot. I have learned your new codes.  In the network_ttn_synthetic.py, I found the sequence is normalized before the trainable layers. For example, if the input is [1,2,3], the output is [2/sqrt(12),2/sqrt(12),2/sqrt(12)].  It seems that this normalization operation transformed the input vector to unit vector and all the elements in the unit vector is same. I wonder whether this normalization operation is necessary for TNN module? To the best of my knowledege, the z-normalization is a common strategy to preprocess the input sequence for time series classification. 

   I don't know what's the best way to normalize the input sequences if I want to use TTN module, maybe a combination of the two or just one of them? I hope to get some advice from you. Thanks for your kind help!
